org.civicrm.reqangular/src/common/services/pub-sub.js

```diff
@@ -0,0 +1,59 @@
+define([
+  'common/modules/services'
+], function (services) {
+  'use strict';
+
+  var topics = Object.create(null);
+
+  /**
+   * This service is used for communication between different angular apps using pub-sub pattern
+   * @ngdoc service
+   * @name pubSub
+   * @requires $rootScope
+   **/
+
+  services.factory("pubSub", ['$rootScope', function ($rootScope) {
+    return {
+      /**
+       * Subscribe method for events
```

Comment:
Again, problem with the documentation. I don't care much about the angular's ngdoc param (we don't use them consistently), but there are three things any method documentation should get right:
1. Description. Should be a good middle ground between being descriptive enough and not being overly detailed
2. Params. All function parameters should be listed with their {type}, name, and if needed description
3. Return value. Should be marked as `@return` and declare the {type} and if necessary the description

With the above said:

```js
/**
 * Subscribe a given listener callback to a specific topic
 *
 * @param {string} topic
 * @param {Function} listener
 * @return {Object} an object that provides the ability to cancel the subscription
 */
subscribe: function (topic, listener) {

}
```